src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -802,6 +802,17 @@ public void MoveToSpecificMethod(IMethodSymbol method, CancellationToken cancell
 
             if (_state._method is not null)
             {
+                // If we didn't have any previous parameters, then there is only the placeholder in the snippet.
+                // We'll use that as text of the first argument; failing to do so means starting to type in the placeholder
+                // will be overwritten once the signature help changes it's overload due to your typing.
```

Comment:
💡 It should be possible to add an integration test for this by typing the following _without waiting_

* `ToString`, <kbd>Tab</kbd>, <kbd>Tab</kbd>, <kbd>0</kbd>

Reply:
I actually have to fix this, because this needs to be conditioned on either the method being not null or also there not being a method at all which is if you hit tab before signature help even caught up.

Reply:
> or also there not being a method at all which is if you hit tab before signature help even caught up

Signature help is only triggered by the _second_ <kbd>Tab</kbd> key, so for the example here only `0` could be typed between this feature getting invoked and signature help appearing.

Reply:
Yeah, that's what I meant. Code is still broken though.

Reply:
So I fixed up the code a bit better, but until we fix or address typing causing model changes, it's hard to really get a sense of what's necessary here. I'd say we fix that, and then sort out a good test for this.